src/PrestaShopBundle/Controller/Admin/Sell/Catalog/SupplierController.php

```diff
@@ -63,8 +68,14 @@
 /**
  * Class SupplierController is responsible for "Sell > Catalog > Brands & Suppliers > Suppliers" page.
  */
-class SupplierController extends FrameworkBundleAdminController
+class SupplierController extends PrestaShopAdminController
 {
+    public function __construct(
+        #[Autowire(service: 'prestashop.core.configuration.ini_configuration')]
+        private readonly IniConfiguration $iniConfig,
```

Comment:
Suggestion (which is valid for other controllers using this service): you could inject this one via `getSubscribedServices` The main advantage being that the service only gets instanciated when you call `$this->container->get(IniConfiguration::class)`

Whereas injecting in the constructor means the service is always instantiated when the controller is instantiated, even when it doesn't use it